client/data/agency-dashboard/use-sync-jetpack-connection-status.ts

```diff
@@ -0,0 +1,61 @@
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { useContext, useEffect, useMemo } from 'react';
+import SitesOverviewContext from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/context';
+import { Site } from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/types';
+import wpcom from 'calypso/lib/wp';
+
+const useSyncJetpackConnectionStatus = (
+	isPartnerOAuthTokenLoaded: boolean,
+	isConnectionHealthy: boolean,
+	siteId: number
+) => {
+	const queryClient = useQueryClient();
+	const { search, currentPage, filter, sort } = useContext( SitesOverviewContext );
+
+	const dashboardSitesQueryKey = useMemo(
+		() => [ 'jetpack-agency-dashboard-sites', search, currentPage, filter, sort ],
```

Comment:
With the generic context in mind, perhaps this key should be named "jetpack-test-connection" or "jetpack-connection-status` instead?

Reply:
This isn't the query key for the custom hook I refactored but the one we're syncing the response to which means if we're adjusting it we should do it in the other query but I think it's name is fitting.